proxy/src/stream.rs

```diff
@@ -228,27 +231,27 @@ impl<S: AsyncRead + AsyncWrite + Unpin> AsyncWrite for Stream<S> {
 }
 
 pin_project! {
-    /// This stream tracks all writes and calls user provided
-    /// callback when the underlying stream is flushed.
-    pub struct MeasuredStream<S, W> {
+    /// This stream tracks all writes, and whenever the stream is flushed,
+    /// increments the user-provided counter by the number of bytes flushed.
+    pub struct MeasuredStream<S> {
         #[pin]
         stream: S,
         write_count: usize,
-        inc_write_count: W,
+        write_counter: prometheus::IntCounter,
     }
 }
 
-impl<S, W> MeasuredStream<S, W> {
-    pub fn new(stream: S, inc_write_count: W) -> Self {
+impl<S> MeasuredStream<S> {
+    pub fn new(stream: S, write_counter: prometheus::IntCounter) -> Self {
```

Comment:
Sorry, this is where we'll draw a line.

For sure, there are good changes in this PR, but I don't like this change, because it's not a refactoring, it's a bikeshed. One of the core benefits of being a project maintainer is the ability to choose and maintain a certain style. What's the point of being a lead if you can't even code the way you enjoy?

Reply:
Ok, so the reason I wanted to simplify the signature of MeasuredStream is that with the other refactorings in this PR, the generic becomes quite cumbersome.

To demonstrate the problem, here's a branch with the MeasuredStream change reverted: https://github.com/neondatabase/neon/tree/heikki/proxy-refactorings-without-measuredstream-simplification. It doesn't compile. I couldn't figure out how to fill in the generic type parameters to make it compile. I'm sure there is a way, but to me the simplest solution seemed to be to remove the unused abstraction. What solution would you prefer?